src/netCDF4/__init__.pyi

```diff
@@ -473,15 +606,14 @@ class Variable(Generic[T_Datatype]):
     def set_ncstring_attrs(self, ncstring_attrs: bool) -> None: ...
     def set_collective(self, value: bool) -> None: ...
     def get_dims(self) -> tuple[Dimension, ...]: ...
-
     def __delattr__(self, name: str) -> None: ...
     def __setattr__(self, name: str, value: Any) -> None: ...
     def __getattr__(self, name: str) -> Any: ...
-    def __getitem__(self, elem: GetSetItemKey) -> np.ndarray: ...
+    def __getitem__(self, elem: GetSetItemKey) -> Any: ...
     def __setitem__(self, elem: GetSetItemKey, data: npt.ArrayLike) -> None: ...
     def __array__(self) -> np.ndarray: ...
     def __len__(self) -> int: ...
-
+    def __iter__(self) -> Iterator[Any]: ...  # faux method so mypy believes Variable is iterable
```

Comment:
Why is this required?

Reply:
Because otherwise you get errors from mypy if you try to iterate over a Variable. [`mypy` doesn't understand that if an object as `__len__` and `__getitem__` then it is iterable](https://stackoverflow.com/questions/61712618/why-does-mypy-not-consider-a-class-as-iterable-if-it-has-len-and-getitem). This came up when I added the `--check-untyped-defs` flag to mypy to look more in-depth into the tests and examples.

Reply:
```suggestion
    def __iter__(self) -> Iterator[np.ma.masked_array]: ...  # faux method so mypy believes Variable is iterable
```

Reply:
Not necessarily. It depends on whether `always_mask` is set (could just return an `ndarray` ), or if it's a string variable then it may just return a string. Same thing as `__getitem__` and `getValue`.